fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/indexes/TextIndexTest.java

```diff
@@ -1039,7 +1043,167 @@ public void saveComplexWithAggressiveConflictRanges() throws Exception {
     }
 
     @Test
-    public void tokenizerVersionChange() throws Exception {
+    void deleteWhereSimpleDocuments() {
+        final SimpleDocument doc1 = SimpleDocument.newBuilder()
+                .setDocId(1623L)
+                .setGroup(0)
+                .setText("foo bar")
+                .build();
+        final SimpleDocument doc2 = SimpleDocument.newBuilder()
+                .setDocId(1945L)
+                .setGroup(0)
+                .setText("foo bar")
+                .build();
+        final RecordMetaDataHook hook = metaDataBuilder -> {
+            TextIndexTestUtils.addRecordTypePrefix(metaDataBuilder);
+            // Add "text" to the simple document primary key so that when we issue a delete where based
+            // on the first field of the SimpleDocument$text index, it will align with the primary key
+            metaDataBuilder.getRecordType(SIMPLE_DOC)
+                    .setPrimaryKey(concat(recordType(), field("text"), field("doc_id")));
+        };
+        try (FDBRecordContext context = openContext()) {
+            openRecordStore(context, hook);
+            recordStore.saveRecord(doc1);
+            recordStore.saveRecord(doc2);
+            commit(context);
+        }
+        try (FDBRecordContext context = openContext()) {
+            openRecordStore(context, hook);
+            Query.InvalidExpressionException err = assertThrows(Query.InvalidExpressionException.class,
+                    () -> recordStore.deleteRecordsWhere(SIMPLE_DOC, Query.field("text").equalsValue("foo bar")));
+            assertThat(err.getMessage(), containsString("deleteRecordsWhere not supported by index SimpleDocument$text"));
+        }
+    }
+
+    @Test
+    void deleteWhereComplexByGroup() {
+        final ComplexDocument zeroGroupDocument = ComplexDocument.newBuilder()
```

Comment:
It seems like a weak test to have only record per group, and only one index entry. It seems like it would be better to have multiple records contain the queried text.

Reply:
Sure, added more records here